apps/dcos_dns/src/dcos_dns_handler.erl

```diff
@@ -151,21 +146,18 @@ resolve_loop(Begin, [], Fun, FailMsg, Zone) ->
 resolve_loop(Begin, Workers, Fun, FailMsg, Zone) ->
     receive
         {reply, Pid, Response} ->
-            {value, {Upstream, Pid, MonRef}, Workers0} =
+            {value, {_Upstream, Pid, MonRef}, Workers0} =
```

Comment:
Would it make sense to use `Upstream` as a metrics tag, since the number of upstream servers is small hence it won't make the metrics cardinality much bigger?

Reply:
If I am not mistaken, the same query for a DNS address (e.g. google.com) could be forwarded to different upstream (eg, 8.8.8.8:53 and 8.8.4.4:53). How would we capture that as a label? 

We do send the `upstream_address` which is an `ip + port` of the Upstream for `worker` metrics (forwarding launches a worker per upstream).  There my guess is that cardinality should not be a problem, as I would expect the number of distinct upstream addresses to be in the order of `10`.